File: Desafio_UFPI.py
turmas = {}  #global




def media(n1,n2,n3):
    media = ((n1*1) + (n2*2) + (n3*3)) / 6
    return f"{media:.2f}"
    
# função menu, exibindo menu
def menu():
    print("**************Bem-Vindo!**************")
    print("*********Menu Controle Acadêmico*********")
    print("1 - Definir Informações da Turma")
    print("2 - Inserir Alunos e Notas")
    print("3 - Exibir Alunos e Médias")
    print("4 - Exibir Alunos Aprovados")
    print("5 - Exibir Alunos Reprovados")
    print("6 - Alterar Notas de Aluno")
    print("7 - Salvar dados em Disco")
    print("8 - Sair do Programa(fim)")
# definindo informações
def definir_Informacoes():
    cod_disciplina = input("Digite o Código da Disciplina: ") 
    
    if cod_disciplina in turmas:
        print("Codigo ja Existe!")
        print("Tente Novamente..")
        sair = input("Tentar Novamente?(Sim/Não): ")
        if sair in ['NÂO','Não','Nao','nao','NAO','N','n']:
            print("Okay Voltando ao Menu Principal!!")
            return
        else: 
            definir_Informacoes()
            return
        
    else:
        nome_disciplina = input("Digite o Nome da Disciplina: ")
        professor = input("Digite o Nome do Professor: ")
        quant_aluno = input("Digite a Quantidade de Aluno: ")
        horario = input("Digite o Horário Inicial: ")
        fim_disciplina = input("Digite o Horário Final: ")
        numero_sala = input("Digite o Número da Sala:  ")
        carga_horaria = input("Digite a Carga Horária: ")
        
        turmas[cod_disciplina] = {'codigo_disci': cod_disciplina,
                                'nome_disciplina': nome_disciplina, 
                                'professor': professor,
                                'quant_aluno': quant_aluno,
                                'horario': horario,
                                'fim_disciplina': fim_disciplina,
                                'numero_sala': numero_sala,
                                'carga_horaria': carga_horaria,
                                'alunos':[]}
        
        
        print(f"""
            Aluno Adicionada com Sucesso!
            Codigo da Disciplina: {cod_disciplina}
            Nome da Disciplina: {nome_disciplina}
            Professor: {professor}
            Quantidade de Aluno: {quant_aluno}
            Horário Inicial: {horario}
            Horário Final: {fim_disciplina}
            Número da Sala: {numero_sala}
            Carga Horária: {carga_horaria}
            """)

def inserir_Alunos():
   
    opc = input("Fazer Busca?\n 1 - Codigo da Disciplina.\n 2 - Nome da Disciplina.\n ")
    if opc == '1':
        cod_disciplina = input("Digite o Codigo da Disciplina: ")
    elif opc == '2':
        nome_disciplina = input("Digite o Nome da Disciplina: ") 
    
        disciplina = False
        for cod, info in turmas.items():
            if info['nome_disciplina'].upper() == nome_disciplina.upper():
                cod_disciplina = cod
                disciplina = True
        if not disciplina:
            print("Disciplina não encontrada. Tente Novamente.")
            sair = input("Tentar Novamente?(Sim/Não): ")
            if sair in ('NÂO','Não','Nao','nao','NAO','N','n'):
                print("Okay Voltando ao Menu Principal!!")
                return
            else: 
                inserir_Alunos()
                return
    else:
        print("Opção Inválida. Tente Novamente.")
        sair = input("Tentar Novamente?(Sim/Não): ")
        if sair in ['NÂO','Não','Nao','nao','NAO','N','n']:
            print("Okay Voltando ao Menu Principal!!")
            return
        else: 
            inserir_Alunos()
            return
            
    
    if cod_disciplina in turmas:
        nome = input("Digite o Nome do Aluno: ")
        n1 = float(input("Digite a Primeira Nota: "))
        n2 = float(input("Digite a Segunda Nota: "))
        n3 = float(input("Digite a Terceira Nota: "))
        
        if n1 < 0 or n1 > 10 or n2 < 0 or n2 > 10 or n3 < 0 or n3 > 10:
            print("Notas devem estar entre 0.0 e 10.0.")
            print("Tente Novamente.")
            inserir_Alunos()
        
        m = media(n1,n2,n3)
        
        turmas[cod_disciplina]['alunos'].append({'nome': nome,'nota1': n1, 'nota2': n2, 'nota3': n3, 'media': m})

        
        print(f"Aluno {nome} Adicionado com Sucesso na Disciplina.\n Codigo: {cod_disciplina}\n Turma: {turmas[cod_disciplina]['nome_disciplina']}")
        print(f"Notas Adicionadas com Sucesso para o Aluno: {nome} Nota 1: {n1} Nota 2: {n2} Nota 3: {n3} ")
        # print(turmas[cod_disciplina]['alunos'])
        
    else:
        print("Codigo da Disciplina Invalido!!")
        print("Tente Novamente.")
        inserir_Alunos()
        
    
def exibir_Media():
  
    opc = input("Fazer Busca?\n 1 - Codigo da Disciplina.\n 2 - Nome da Disciplina.\n ")
    if opc == '1':
        cod_disciplina = input("Digite o Codigo da Disciplina: ")
    elif opc == '2':
        nome_disciplina = input("Digite o Nome da Disciplina: ") 
    
        disciplina = False
        for cod, info in turmas.items():
            if info['nome_disciplina'].upper() == nome_disciplina.upper():
                cod_disciplina = cod
                disciplina = True
        if not disciplina:
            print("Disciplina não encontrada. Tente Novamente.")
            sair = input("Tentar Novamente?(Sim/Não): ")
            if sair in ('NÂO','Não','Nao','nao','NAO','N','n'):
                print("Okay Voltando ao Menu Principal!!")
                return
            else: 
                exibir_Media()
                return
    else:
        print("Opção Inválida. Tente Novamente.")
        sair = input("Tentar Novamente?(Sim/Não): ")
        if sair in ['NÂO','Não','Nao','nao','NAO','N','n']:
            print("Okay Voltando ao Menu Principal!!")
            return
        else: 
            exibir_Media()
            return
    
    if cod_disciplina in turmas:
        for aluno in turmas[cod_disciplina]['alunos']:
            print(f"Aluno: {aluno['nome']}, Média: {aluno['media']}")
    else:
        print("Codigo da Disciplina Inválido! ")
        print("Tente Novamente. ")
        sair = input("Tentar Novamente?(Sim/Não): ")
        if sair in ['NÂO','Não','Nao','nao','NAO','N','n']:
            print("Okay Voltando ao Menu Principal!!")
            return
        else: 
            exibir_Media()
            return
        

def alunos_Aprovados():
    opc = input("Fazer Busca?\n 1 - Codigo da Disciplina.\n 2 - Nome da Disciplina.\n ")
    if opc == '1':
        cod_disciplina = input("Digite o Codigo da Disciplina: ")
    elif opc == '2':
        nome_disciplina = input("Digite o Nome da Disciplina: ") 
    
        disciplina = False
        for cod, info in turmas.items():
            if info['nome_disciplina'].upper() == nome_disciplina.upper():
                cod_disciplina = cod
                disciplina = True
        if not disciplina:
            print("Disciplina não encontrada. Tente Novamente.")
            sair = input("Tentar Novamente?(Sim/Não): ")
            if sair in ('NÂO','Não','Nao','nao','NAO','N','n'):
                print("Okay Voltando ao Menu Principal!!")
                return
            else: 
                alunos_Aprovados()
                return
    else:
        print("Opção Inválida. Tente Novamente.")
        sair = input("Tentar Novamente?(Sim/Não): ")
        if sair in ['NÂO','Não','Nao','nao','NAO','N','n']:
            print("Okay Voltando ao Menu Principal!!")
            return
        else: 
            alunos_Aprovados()
            return
        
    if cod_disciplina in turmas:
        print(f"Alunos Aprovados na Disciplina {cod_disciplina}")
        for aluno in turmas[cod_disciplina]['alunos']:
            if float(aluno['media']) >= 7.0:
                print(f"Aluno: {aluno['nome']}, Média: {aluno['media']}")
    else:
        print("Codigo da Disciplina")
        print("Tente Novamente.")
        sair = input("Tentar Novamente?(Sim/Não): ")
        if sair in ['NÂO','Não','Nao','nao','NAO','N','n']:
            print("Okay Voltando ao Menu Principal!!")
            return
        else: 
            alunos_Aprovados()
            return
        
        
        
def alunos_Reprovados():
    opc = input("Fazer Busca?\n 1 - Codigo da Disciplina.\n 2 - Nome da Disciplina.\n ")
    if opc == '1':
        cod_disciplina = input("Digite o Codigo da Disciplina: ")
    elif opc == '2':
        nome_disciplina = input("Digite o Nome da Disciplina: ") 
    
        disciplina = False
        for cod, info in turmas.items():
            if info['nome_disciplina'].upper() == nome_disciplina.upper():
                cod_disciplina = cod
                disciplina = True
        if not disciplina:
            print("Disciplina não encontrada. Tente Novamente.")
            sair = input("Tentar Novamente?(Sim/Não): ")
            if sair in ('NÂO','Não','Nao','nao','NAO','N','n'):
                print("Okay Voltando ao Menu Principal!!")
                return
            else: 
                alunos_Reprovados()
                return
    else:
        print("Opção Inválida. Tente Novamente.")
        sair = input("Tentar Novamente?(Sim/Não): ")
        if sair in ['NÂO','Não','Nao','nao','NAO','N','n']:
            print("Okay Voltando ao Menu Principal!!")
            return
        else: 
            alunos_Reprovados()
            return
        
    if cod_disciplina in turmas:
        print(f"Alunos Aprovados na Disciplina {cod_disciplina}")
        for aluno in turmas[cod_disciplina]['alunos']:
            if float(aluno['media']) < 7.0:
                print(f"Aluno: {aluno['nome']}, Média: {aluno['media']}")
    else:
        print("Codigo da Disciplina")
        print("Tente Novamente.")
        sair = input("Tentar Novamente?(Sim/Não): ")
        if sair in ['NÂO','Não','Nao','nao','NAO','N','n']:
            print("Okay Voltando ao Menu Principal!!")
            return
        else: 
            alunos_Reprovados()
            return
        

def alterar_Notas_alunos():
    opc = input("Fazer Busca?\n 1 - Codigo da Disciplina.\n 2 - Nome da Disciplina.\n ")
    if opc == '1':
        cod_disciplina = input("Digite o Codigo da Disciplina: ")
    elif opc == '2':
        nome_disciplina = input("Digite o Nome da Disciplina: ") 
    
        disciplina = False
        for cod, info in turmas.items():
            if info['nome_disciplina'].upper() == nome_disciplina.upper():
                cod_disciplina = cod
                disciplina = True
        if not disciplina:
            print("Disciplina não encontrada. Tente Novamente.")
            sair = input("Tentar Novamente?(Sim/Não): ")
            if sair in ('NÂO','Não','Nao','nao','NAO','N','n'):
                print("Okay Voltando ao Menu Principal!!")
                return
            else: 
                alterar_Notas_alunos()
                return
    else:
        print("Opção Inválida. Tente Novamente.")
        sair = input("Tentar Novamente?(Sim/Não): ")
        if sair in ['NÂO','Não','Nao','nao','NAO','N','n']:
            print("Okay Voltando ao Menu Principal!!")
            return
        else: 
            alterar_Notas_alunos()
            return
        
        
    if cod_disciplina in turmas:
        for aluno in turmas[cod_disciplina]['alunos']:
            print(f"Aluno: {aluno['nome']}")
        escolha_A = input("Digite o nome do Aluno para Mudar a Nota: ")

        for aluno in turmas[cod_disciplina]['alunos']:
            if aluno['nome'].lower() == escolha_A.lower():
                print(f"Aluno: {escolha_A}, {aluno['nota1']}, {aluno['nota2']}, {aluno['nota3']}")
                n = input("""Escolha:
                          1 - Nota 1
                          2 - Nota 2
                          3 - Nota 3
                          Para alterar a Nota: """)
                nova_n = input("Digite a Nova Nota: ")
                aluno['nota'+n] = nova_n
            else:
                print("Aluno não encontrado!!")
                alterar_Notas_alunos()
    
    else:
        print("Codigo da Disciplina Inválido!")
        print("Tente Novamente.")
        sair = input("Tentar Novamente?(Sim/Não): ")
        if sair in ['NÂO','Não','Nao','nao','NAO','N','n']:
            print("Okay Voltando ao Menu Principal!!")
            return
        else: 
            alterar_Notas_alunos()
            return
        


def salvar_dados():
    nome_arquivo = input("Digite o nome do arquivo para salvar os dados (sem extensão): ") + '.txt'
    
    try:
        with open(nome_arquivo, 'w') as arquivo:
            for cod_disciplina, info_disciplina in turmas.items():
                arquivo.write(f"Código da Disciplina: {info_disciplina['codigo_disci']}\n")
                arquivo.write(f"Nome da Disciplina: {info_disciplina['nome_disciplina']}\n")
                arquivo.write(f"Professor: {info_disciplina['professor']}\n")
                arquivo.write(f"Quantidade de Alunos: {info_disciplina['quant_aluno']}\n")
                arquivo.write(f"Horário Inicial: {info_disciplina['horario']}\n")
                arquivo.write(f"Horário Final: {info_disciplina['fim_disciplina']}\n")
                arquivo.write(f"Número da Sala: {info_disciplina['numero_sala']}\n")
                arquivo.write(f"Carga Horária: {info_disciplina['carga_horaria']}\n")
                arquivo.write("Alunos:\n")
                
                for aluno in info_disciplina['alunos']:
                    arquivo.write(f"    Nome: {aluno['nome']}, Nota1: {aluno['nota1']}, Nota2: {aluno['nota2']}, Nota3: {aluno['nota3']}, Média: {aluno['media']}\n")
                
                arquivo.write("\n")
        
        print(f"Dados salvos com sucesso no arquivo {nome_arquivo}")
    
    except IOError:
        print(f"Erro ao salvar dados no arquivo {nome_arquivo}")

def sair():
    print("Encerrando o Programa.... Tchau!!Tchau!!")
    quit()


def main():

    while True:
        menu()
        opc = input("Escolha uma Opção! ")
        
        if opc == '1' :
            definir_Informacoes()
        elif opc == '2' :
            inserir_Alunos()
        elif opc == '3' :
            exibir_Media()
        elif opc == '4' :
            alunos_Aprovados()
        elif opc == '5' :
            alunos_Reprovados()
        elif opc == '6' :
            alterar_Notas_alunos()
        elif opc == '7' :
            salvar_dados()
        elif opc == '8' :
            sair()
            # print("Encerrando o Programa... Tchau!!")
            # break
        else:
            print("Opção Inválida. Escolha Novamente.")
    
if __name__ == "__main__":
    main()
    

File: DesafioUFPI.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Estrutura para armazenar informações da disciplina
struct Disciplina {
    char codigo_disci[20];
    char nome_disciplina[100];
    char professor[100];
    int quant_aluno;
    char horario[20];
    char fim_disciplina[20];
    int numero_sala;
    int carga_horaria;
    struct Aluno *alunos;
    int total_alunos;
};

// Estrutura para armazenar informações do aluno
struct Aluno {
    char nome[100];
    float nota1;
    float nota2;
    float nota3;
    float media;
};

// Protótipos das funções
void menu();
void definir_Informacoes(struct Disciplina *turmas, int *total_turmas);
void inserir_Alunos(struct Disciplina *turmas, int total_turmas);
void exibir_Media(struct Disciplina *turmas, int total_turmas);
void alunos_Aprovados(struct Disciplina *turmas, int total_turmas);
void alunos_Reprovados(struct Disciplina *turmas, int total_turmas);
void alterar_Notas_alunos(struct Disciplina *turmas, int total_turmas);
void salvar_dados(struct Disciplina *turmas, int total_turmas);
void sair();

int main() {
    struct Disciplina turmas[100]; // Array para armazenar até 100 disciplinas
    int total_turmas = 0; // Contador de disciplinas

    while (1) {
        menu();
        int opc;
        printf("Escolha uma Opção: ");
        scanf("%d", &opc);

        switch (opc) {
            case 1:
                definir_Informacoes(turmas, &total_turmas);
                break;
            case 2:
                inserir_Alunos(turmas, total_turmas);
                break;
            case 3:
                exibir_Media(turmas, total_turmas);
                break;
            case 4:
                alunos_Aprovados(turmas, total_turmas);
                break;
            case 5:
                alunos_Reprovados(turmas, total_turmas);
                break;
            case 6:
                alterar_Notas_alunos(turmas, total_turmas);
                break;
            case 7:
                salvar_dados(turmas, total_turmas);
                break;
            case 8:
                sair();
                break;
            default:
                printf("Opção Inválida. Escolha Novamente.\n");
                break;
        }
    }

    return 0;
}

// Função para exibir o menu
void menu() {
    printf("**************Bem-Vindo!**************\n");
    printf("*********Menu Controle Acadêmico*********\n");
    printf("1 - Definir Informações da Turma\n");
    printf("2 - Inserir Alunos e Notas\n");
    printf("3 - Exibir Alunos e Médias\n");
    printf("4 - Exibir Alunos Aprovados\n");
    printf("5 - Exibir Alunos Reprovados\n");
    printf("6 - Alterar Notas de Aluno\n");
    printf("7 - Salvar dados em Disco\n");
    printf("8 - Sair do Programa\n");
}

// Função para definir informações da turma
void definir_Informacoes(struct Disciplina *turmas, int *total_turmas) {
    if (*total_turmas >= 100) {
        printf("Limite de disciplinas atingido!\n");
        return;
    }

    printf("Digite o Código da Disciplina: ");
    scanf(" %s", turmas[*total_turmas].codigo_disci);

    // Verifica se o código da disciplina já existe
    for (int i = 0; i < *total_turmas; i++) {
        if (strcmp(turmas[i].codigo_disci, turmas[*total_turmas].codigo_disci) == 0) {
            printf("Código já existe!\n");
            return;
        }
    }

    printf("Digite o Nome da Disciplina: ");
    scanf(" %[^\n]s", turmas[*total_turmas].nome_disciplina);

    printf("Digite o Nome do Professor: ");
    scanf(" %[^\n]s", turmas[*total_turmas].professor);

    printf("Digite a Quantidade de Alunos: ");
    scanf("%d", &turmas[*total_turmas].quant_aluno);

    printf("Digite o Horário Inicial: ");
    scanf(" %[^\n]s", turmas[*total_turmas].horario);

    printf("Digite o Horário Final: ");
    scanf(" %[^\n]s", turmas[*total_turmas].fim_disciplina);

    printf("Digite o Número da Sala: ");
    scanf("%d", &turmas[*total_turmas].numero_sala);

    printf("Digite a Carga Horária: ");
    scanf("%d", &turmas[*total_turmas].carga_horaria);

    // Aloca memória para o array de alunos da disciplina
    turmas[*total_turmas].alunos = malloc(turmas[*total_turmas].quant_aluno * sizeof(struct Aluno));
    turmas[*total_turmas].total_alunos = 0;

    (*total_turmas)++;

    printf("Disciplina Adicionada com Sucesso!\n");
}

// Função para inserir alunos e suas notas
void inserir_Alunos(struct Disciplina *turmas, int total_turmas) {
    char cod_disciplina[20];
    printf("Digite o Código da Disciplina: ");
    scanf(" %s", cod_disciplina);

    int disciplina_encontrada = 0;
    for (int i = 0; i < total_turmas; i++) {
        if (strcmp(turmas[i].codigo_disci, cod_disciplina) == 0) {
            disciplina_encontrada = 1;

            if (turmas[i].total_alunos < turmas[i].quant_aluno) {
                printf("Digite o Nome do Aluno: ");
                scanf(" %[^\n]s", turmas[i].alunos[turmas[i].total_alunos].nome);

                printf("Digite a Primeira Nota: ");
                scanf("%f", &turmas[i].alunos[turmas[i].total_alunos].nota1);

                printf("Digite a Segunda Nota: ");
                scanf("%f", &turmas[i].alunos[turmas[i].total_alunos].nota2);

                printf("Digite a Terceira Nota: ");
                scanf("%f", &turmas[i].alunos[turmas[i].total_alunos].nota3);

                // Calcula a média do aluno
                turmas[i].alunos[turmas[i].total_alunos].media = (turmas[i].alunos[turmas[i].total_alunos].nota1 +
                                                                  2 * turmas[i].alunos[turmas[i].total_alunos].nota2 +
                                                                  3 * turmas[i].alunos[turmas[i].total_alunos].nota3) / 6.0;

                turmas[i].total_alunos++;

                printf("Aluno adicionado com sucesso na disciplina.\n");
            } else {
                printf("Limite de alunos atingido para esta disciplina.\n");
            }

            break;
        }
    }

    if (!disciplina_encontrada) {
        printf("Disciplina não encontrada.\n");
    }
}

// Função para exibir as médias dos alunos de uma disciplina
void exibir_Media(struct Disciplina *turmas, int total_turmas) {
    char cod_disciplina[20];
    printf("Digite o Código da Disciplina: ");
    scanf(" %s", cod_disciplina);

    int disciplina_encontrada = 0;
    for (int i = 0; i < total_turmas; i++) {
        if (strcmp(turmas[i].codigo_disci, cod_disciplina) == 0) {
            disciplina_encontrada = 1;

            printf("Alunos e suas Médias:\n");
            for (int j = 0; j < turmas[i].total_alunos; j++) {
                printf("Aluno: %s, Média: %.2f\n", turmas[i].alunos[j].nome, turmas[i].alunos[j].media);
            }

            break;
        }
    }

    if (!disciplina_encontrada) {
        printf("Disciplina não encontrada.\n");
    }
}

// Função para exibir os alunos aprovados em uma disciplina
void alunos_Aprovados(struct Disciplina *turmas, int total_turmas) {
    char cod_disciplina[20];
    printf("Digite o Código da Disciplina: ");
    scanf(" %s", cod_disciplina);

    int disciplina_encontrada = 0;
    for (int i = 0; i < total_turmas; i++) {
        if (strcmp(turmas[i].codigo_disci, cod_disciplina) == 0) {
            disciplina_encontrada = 1;

            printf("Alunos Aprovados na Disciplina:\n");
            for (int j = 0; j < turmas[i].total_alunos; j++) {
                if (turmas[i].alunos[j].media >= 7.0) {
                    printf("Aluno: %s, Média: %.2f\n", turmas[i].alunos[j].nome, turmas[i].alunos[j].media);
                }
            }

            break;
        }
    }

    if (!disciplina_encontrada) {
        printf("Disciplina não encontrada.\n");
    }
}

// Função para exibir os alunos reprovados em uma disciplina
void alunos_Reprovados(struct Disciplina *turmas, int total_turmas) {
    char cod_disciplina[20];
    printf("Digite o Código da Disciplina: ");
    scanf(" %s", cod_disciplina);

    int disciplina_encontrada = 0;
    for (int i = 0; i < total_turmas; i++) {
        if (strcmp(turmas[i].codigo_disci, cod_disciplina) == 0) {
            disciplina_encontrada = 1;

            printf("Alunos Reprovados na Disciplina:\n");
            for (int j = 0; j < turmas[i].total_alunos; j++) {
                if (turmas[i].alunos[j].media < 7.0) {
                    printf("Aluno: %s, Média: %.2f\n", turmas[i].alunos[j].nome, turmas[i].alunos[j].media);
                }
            }

            break;
        }
    }

    if (!disciplina_encontrada) {
        printf("Disciplina não encontrada.\n");
    }
}

// Função para alterar as notas de um aluno
void alterar_Notas_alunos(struct Disciplina *turmas, int total_turmas) {
    char cod_disciplina[20];
    printf("Digite o Código da Disciplina: ");
    scanf(" %s", cod_disciplina);

    int disciplina_encontrada = 0;
    for (int i = 0; i < total_turmas; i++) {
        if (strcmp(turmas[i].codigo_disci, cod_disciplina) == 0) {
            disciplina_encontrada = 1;

            printf("Alunos da Disciplina %s:\n", turmas[i].nome_disciplina);
            for (int j = 0; j < turmas[i].total_alunos; j++) {
                printf("%d - %s\n", j + 1, turmas[i].alunos[j].nome);
            }

            char escolha_A[100];
            printf("Digite o nome do Aluno para Mudar a Nota: ");
            scanf(" %[^\n]s", escolha_A);

            int aluno_encontrado = 0;
            for (int j = 0; j < turmas[i].total_alunos; j++) {
                if (strcmp(turmas[i].alunos[j].nome, escolha_A) == 0) {
                    aluno_encontrado = 1;

                    printf("Aluno: %s, Nota1: %.2f, Nota2: %.2f, Nota3: %.2f\n",
                           turmas[i].alunos[j].nome,
                           turmas[i].alunos[j].nota1,
                           turmas[i].alunos[j].nota2,
                           turmas[i].alunos[j].nota3);

                    int n;
                    printf("Escolha:\n");
                    printf("1 - Nota 1\n");
                    printf("2 - Nota 2\n");
                    printf("3 - Nota 3\n");
                    printf("Para alterar a Nota: ");
                    scanf("%d", &n);

                    float nova_n;
                    printf("Digite a Nova Nota: ");
                    scanf("%f", &nova_n);

                    switch (n) {
                        case 1:
                            turmas[i].alunos[j].nota1 = nova_n;
                            break;
                        case 2:
                            turmas[i].alunos[j].nota2 = nova_n;
                            break;
                        case 3:
                            turmas[i].alunos[j].nota3 = nova_n;
                            break;
                        default:
                            printf("Opção Inválida.\n");
                            break;
                    }

                    // Atualiza a média do aluno
                    turmas[i].alunos[j].media = (turmas[i].alunos[j].nota1 +
                                                 2 * turmas[i].alunos[j].nota2 +
                                                 3 * turmas[i].alunos[j].nota3) / 6.0;

                    printf("Nota alterada com sucesso.\n");
                    break;
                }
            }

            if (!aluno_encontrado) {
                printf("Aluno não encontrado.\n");
            }

            break;
        }
    }

    if (!disciplina_encontrada) {
        printf("Disciplina não encontrada.\n");
    }
}

// Função para salvar os dados em disco
void salvar_dados(struct Disciplina *turmas, int total_turmas) {
    char nome_arquivo[100];
    printf("Digite o nome do arquivo para salvar os dados (sem extensão): ");
    scanf(" %[^\n]s", nome_arquivo);
    strcat(nome_arquivo, ".txt");

    FILE *arquivo;
    arquivo = fopen(nome_arquivo, "w");

    if (arquivo == NULL) {
        printf("Erro ao criar arquivo.\n");
        return;
    }

    for (int i = 0; i < total_turmas; i++) {
        fprintf(arquivo, "Código da Disciplina: %s\n", turmas[i].codigo_disci);
        fprintf(arquivo, "Nome da Disciplina: %s\n", turmas[i].nome_disciplina);
        fprintf(arquivo, "Professor: %s\n", turmas[i].professor);
        fprintf(arquivo, "Quantidade de Alunos: %d\n", turmas[i].quant_aluno);
        fprintf(arquivo, "Horário Inicial: %s\n", turmas[i].horario);
        fprintf(arquivo, "Horário Final: %s\n", turmas[i].fim_disciplina);
        fprintf(arquivo, "Número da Sala: %d\n", turmas[i].numero_sala);
        fprintf(arquivo, "Carga Horária: %d\n", turmas[i].carga_horaria);
        fprintf(arquivo, "Alunos:\n");

        for (int j = 0; j < turmas[i].total_alunos; j++) {
            fprintf(arquivo, "    Nome: %s, Nota1: %.2f, Nota2: %.2f, Nota3: %.2f, Média: %.2f\n",
                    turmas[i].alunos[j].nome,
                    turmas[i].alunos[j].nota1,
                    turmas[i].alunos[j].nota2,
                    turmas[i].alunos[j].nota3,
                    turmas[i].alunos[j].media);
        }

        fprintf(arquivo, "\n");
    }

    fclose(arquivo);
    printf("Dados salvos com sucesso no arquivo %s\n", nome_arquivo);
}

// Função para encerrar o programa
void sair() {
    printf("Encerrando o Programa.... Tchau!!Tchau!!\n");
    exit(0);
}
